plugins/catalog-backend/src/service/request/parseEntityTransformParams.ts

```diff
@@ -20,6 +20,22 @@ import lodash from 'lodash';
 import { RecursivePartial } from '../../util/RecursivePartial';
 import { parseStringsParam } from './common';
 
+function getPathArray(input: Entity, field: string) {
+  const pathArray = [];
+  let currentPathPart = '';
+
+  for (const pathPart of field.split('.')) {
+    currentPathPart += pathPart;
+
+    if (lodash.has(input, pathArray.concat(currentPathPart))) {
+      pathArray.push(currentPathPart);
+      currentPathPart = '';
+    }
+  }
+
+  return pathArray;
+}
```

Comment:
This has bugs - the test succeeds "by accident" since it'll stop at the annotations level and return that, and the example input data there only had one annotation. I think the problem is, your currentPathPart doesn't add a dot between parts.

While you are in this function, it could actually have kept track of the current level of the hierarchy and returned the value it found straight away (so the return type could be `[path: string[], value: unknown]` or so). That'd probably be significantly more efficient than constantly doing `lodash.has` and re-traversing down the hierarchy over and over at every level.

Reply:
Ok. I'll make that change.

Reply:
@freben updated to fix the bug and add more tests. 

Will come back and change the implementation to what you suggested, just wanted to first correct this implementation and add the additional tests.